### PR TITLE
Switch to pre-built Cassandra Docker image

### PR DESCRIPTION
**Pull request by rkargMsft**
Removed `Cassandra.dockerfile` and related custom image build code. Updated `CassandraContainer.cs` to use a pre-built Cassandra image from Docker Hub, specified by the `CASSANDRAVERSION` environment variable.

Using the base image without the prior config changes increases a local run of all the tests from 18 sec. to about 70 sec. Annoying, but not terrible.

We could look at replacing the config file with a transformed version but I don't know how stable that would be with future versions of Cassandra.

The test run on the build server should let us know if this is acceptable speed there.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/orleans/pull/9076)

### CODE REVIEW

**Comment by rkargMsft** [comment]
Adds about a minute to the test run on the build server, too.

**Comment by rkargMsft** [comment]
Test failures appear to be in tests unrelated to the changes in this PR

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -86,7 +86,7 @@ jobs:
           3.1.x
           8.0.x
     - name: Test
-      run: dotnet test --filter "Category=${{ matrix.provider }}&(Category=BVT|Category=SlowBVT|Category=Functional)" --blame-hang-timeout 10m --logger "trx" -- -parallel none -noshadow
+      run: dotnet test --filter "Category=${{ matrix.provider }}&(Category=BVT|Category=SlowBVT|Category=CLustering)" --blame-hang-timeout 10m --logger "trx" -- -parallel none -noshadow
```
Comment (ReubenBond) [inline .github/workflows/ci.yml:89]:
```suggestion
      run: dotnet test --filter "Category=${{ matrix.provider }}&(Category=BVT|Category=SlowBVT|Category=Clustering)" --blame-hang-timeout 10m --logger "trx" -- -parallel none -noshadow
```

**Comment by benjaminpetit** [comment]
Thanks!

**Pull request merged**